descartes_planner/src/planning_graph.cpp

```diff
@@ -763,95 +761,53 @@ void PlanningGraph::printGraph()
-        JointTrajectoryPt new_pt (*joint_pose_iter, trajectory_iter->second.source_trajectory_.get()->getTiming());
-        traj_solutions.push_back(new_pt.getID());
-        joint_solutions_map_[new_pt.getID()] = new_pt;
-      }
+      poses[i].emplace_back(std::move(sol), points[i]->getTiming());
     }
-    trajectory_iter->second.joints_ = traj_solutions;
-    /*************************/
   }
 
   return true;
 }
 
-bool PlanningGraph::calculateAllEdgeWeights(std::vector<JointEdge> &edges)
+bool PlanningGraph::calculateAllEdgeWeights(const std::vector<std::vector<JointTrajectoryPt>>& poses, std::vector<JointEdge> &edges)
 {
-  if (cartesian_point_link_->size() == 0)
-  {
-    // no linkings of cartesian points
-    ROS_ERROR_STREAM("no trajectory point links defined");
-    return false;
-  }
-  else
-  {
-    ROS_DEBUG_STREAM("Found %i " << cartesian_point_link_->size() << " trajectory point links");
-  }
-
-  if (joint_solutions_map_.size() == 0)
-  {
-    // no joint solutions to calculate transitions for
-    ROS_ERROR_STREAM("no joint solutions available");
-    return false;
-  }
-  else
-  {
-    ROS_INFO_STREAM("Calculating edges for " << joint_solutions_map_.size() << " total joint solutions");
-  }
-
-  for (auto cart_link_iter = cartesian_point_link_->begin();
-      cart_link_iter != cartesian_point_link_->end(); ++cart_link_iter)
+  // We check that the size of input traj is at least 2 at the start of insertGraph()
+  // iterate over each pair of points
+  for (std::size_t i = 1; i < poses.size(); ++i)
   {
-    TrajectoryPt::ID start_cart_id = cart_link_iter->first; // should be equal to cart_link_iter->second.links_.id
-    TrajectoryPt::ID end_cart_id = cart_link_iter->second.links_.id_next;
+    const std::vector<JointTrajectoryPt>& from = poses[i - 1];
+    const std::vector<JointTrajectoryPt>& to = poses[i];
 
-    if(!end_cart_id.is_nil())
+    if (!calculateEdgeWeights(from, to, edges))
     {
-      std::vector<TrajectoryPt::ID> start_joint_ids = cart_link_iter->second.joints_;
-      std::vector<TrajectoryPt::ID> end_joint_ids = (*cartesian_point_link_)[end_cart_id].joints_;
-
-      if(!calculateEdgeWeights(start_joint_ids, end_joint_ids, edges))
-      {
-        ROS_WARN_STREAM("One or more joints lists in the cartesian point link is empty ID:" << start_cart_id << 
-                        "[start ids:" << start_joint_ids.size() << "], ID:" << end_cart_id << "[end ids:" << end_joint_ids.size() << ']');
-      }
+      ROS_ERROR_STREAM(__FUNCTION__ << ": unable to calculate any valid transitions between inputs " << (i-1) << " and " << i); 
+      return false;
     }
   }
 
   return !edges.empty();
 }
 
-bool PlanningGraph::calculateEdgeWeights(const std::vector<TrajectoryPt::ID> &start_joints,const std::vector<TrajectoryPt::ID> &end_joints, std::vector<JointEdge> &edge_results)
+bool PlanningGraph::calculateEdgeWeights(const std::vector<TrajectoryPt::ID> &start_joints,
```

Comment:
Since this method might potentially be deprecated we should log it as an issue so that it's not forgotten.